src/main/java/network/brightspots/rcv/ContestConfigMigration.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Universal RCV Tabulator
+ * Copyright (c) 2017-2020 Bright Spots Developers.
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+ * the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this
+ * program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package network.brightspots.rcv;
+
+import static network.brightspots.rcv.Utils.isNullOrBlank;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import network.brightspots.rcv.RawContestConfig.ContestRules;
+import network.brightspots.rcv.RawContestConfig.CvrSource;
+import network.brightspots.rcv.Tabulator.OvervoteRule;
+import network.brightspots.rcv.Tabulator.TieBreakMode;
+import network.brightspots.rcv.Tabulator.WinnerElectionMode;
+
+class ContestConfigMigration {
+
+  private static final Pattern versionNumPattern = Pattern.compile("(\\d+).*");
+
+  private static ArrayList<Integer> parseVersionString(String version) {
+    ArrayList<Integer> parsed = new ArrayList<>();
+    String[] arr = version.split("\\.");
+    for (String numString : arr) {
+      Matcher m = versionNumPattern.matcher(numString);
+      if (m.matches()) {
+        parsed.add(Integer.parseInt(m.group(1)));
+      }
+    }
+
+    return parsed;
+  }
+
+  // not intended to be used if either version is null
+  private static boolean isVersionNewer(String version1, String version2) {
+    if (version1.equals(version2)) {
+      return false;
```

Comment:
There's a bunch of early returns in this file which sadly we are not allowed to use.

Reply:
sigh